### ISSUE TITLE
Parser always unwraps spans, no matter the rules.

### ISSUE THREAD

**Issue by mchanton** [Voog/wysihtml#317]
I have HTML like this:
<p>
<span begin="xxx" end="xxx">Hi</span> <span begin="xxx" end="xxx">how</span> <span begin="xxx" end="xxx">are</span> <span begin="xxx" end="xxx">you</span>
</p>

The editor always unwraps these span elements like this:

<p>
Hi how are you
</p>

This also happens with spans that don't have the "begin" or "end" attributes.

This happens with all premade rulesets. (like 'advanced_unwrap' or 'simple'). It also happens if I create a custom rule for spans like this: 

span: 1,

If I substitute the span tags with 'em' or 's', the rules apply like I want to. I don't get why spans are treated differently than 's' or 'em' elements.

**Comment by pulges** [comment]
Will it help if you set this option to false?
https://github.com/Voog/wysihtml/wiki/Configuration#cleanup

**Comment by mchanton** [comment]
That actually fixed it, thanks!
I think that option shouldn't cleanup spans with attributes, though. Especially if you allow those attributes in the rules.

**Comment by pulges** [comment]
You are correct. The current cleanup code is a bit ambitious, not really counting for attributes. Will try to look at it if I get more time to focus.

I think it is this line where the problem is, if anyone would like to give it a try:
https://github.com/Voog/wysihtml/blob/master/src/dom/parse.js#L223
